test/inProgressRepositoryService.test.ts

```diff
@@ -2,932 +2,1592 @@ import IAlternative from '@shared/interface/IAlternative';
+          description: 'description',
+          isFavorable: false,
+          dataSources: [
+            {
+              id: 'ds2Id',
+              source: 'reference',
+              unitOfMeasurement: {
+                type: UnitOfMeasurementType.percentage,
+                label: '%'
+              },
+              strengthOfEvidence: 'strengths',
+              uncertainties: 'uncertainties',
+              scale: [0, 100]
+            }
+          ]
         }
-      },
-      distributions: {
-        ds1Id: {
-          alt1Id: {} as Distribution
+      };
+      const alternatives: Record<string, {title: string}> = {
+        alt1Id: {
+          title: 'alternative'
         }
-      }
-    };
-    try {
-      createProblem(inProgressMessage);
-    } catch (error) {
-      expect(error).toBe('Cell without effect and distribution found');
-    }
+      };
+
+      const result = createOrdering(criteria, alternatives);
+
+      const expectedResult: IOrdering = {
+        criteria: ['crit1Id', 'crit2Id'],
+        alternatives: [alternative1Id],
+        dataSources: [dataSource1Id, 'ds2Id']
+      };
+
+      expect(result).toEqual(expectedResult);
+    });
   });
-});
 
-describe('createOrdering', () => {
-  it('should create ordering', () => {
-    const criteria: Record<string, IProblemCriterion> = {
-      crit1Id: {
-        title: 'title',
-        description: 'description',
-        isFavorable: false,
-        dataSources: [
+  describe('buildEmptyInProgress', () => {
+    it('should return a new workspace ready to insert into the database', () => {
+      const result = buildEmptyInProgress();
+      const expectedWorkspace = {
+        title: 'new workspace',
+        therapeuticContext: '',
+        useFavourability: true
+      };
+      expect(result.workspace).toEqual(expectedWorkspace);
+
+      const expectedCriterion = {
+        isFavourable: true,
+        title: `criterion 1`,
+        description: ''
+      };
+      expect(_.omit(result.criteria[0], ['id', 'dataSources'])).toEqual(
+        expectedCriterion
+      );
+
+      const expectedDataSource = {
+        reference: '',
+        uncertainty: '',
+        strengthOfEvidence: '',
+        unitOfMeasurement: {
+          label: '',
+          type: UnitOfMeasurementType.custom
+        }
+      };
+      expect(
+        _.omit(result.criteria[0].dataSources[0], ['id', 'criterionId'])
+      ).toEqual(expectedDataSource);
+
+      expect(result.criteria[1].title).toEqual('criterion 2');
+      expect(result.alternatives[0].title).toEqual('alternative 1');
+      expect(result.alternatives[1].title).toEqual('alternative 2');
+      expect(result.effects).toEqual([]);
+      expect(result.distributions).toEqual([]);
+    });
+  });
+
+  describe('buildInProgressCopy', () => {
+    const problem: IProblem = {
+      schemaVersion: 'some version',
+      performanceTable: [],
+      description: '',
+      title: 'my workspace',
+      alternatives: {},
```

Comment:
a comment here to the effect that the alternatives/crits are being tested further below would help